projects/components/src/color-picker/color-picker.component.ts

```diff
@@ -49,6 +49,10 @@ export class ColorPickerComponent implements ControlValueAccessor, OnChanges {
   @Input()
   public disabled?: boolean = false;
 
+  // Disables de-selecting current option, meaning only allow selection to another option
+  @Input()
+  public disableDeselection?: boolean = false;
```

Comment:
we don't need making it optional, if we have assigned it a value

Reply:
what a name! how about `required` ?

Reply:
The reason we often make it optional with an assigned value is to support async assignments. Since angular won't prevent us from writing something like
`[disableDeselection] = "isRequired$ | async"`, that will result in a null value being assigned and overwriting the default. So the optional - even though it's not null, which would be more accurate - makes sure our code doesn't break if that happens.

Reply:
I like `disableDeselection` because this case seems a bit different than the standard required form case, however nbd to me, i can change it if you like.

> The reason we often make it optional with an assigned value is to support async assignments. Since angular won't prevent us from writing something like
[disableDeselection] = "isRequired$ | async", that will result in a null value being assigned and overwriting the default. So the optional - even though it's not null, which would be more accurate - makes sure our code doesn't break if that happens.

Thats a good point, and safety measure, i'll change back.

Reply:
If deselection is not allowed, can we disable the color picker after selecting a value?

Reply:
No because we still want to allow them to select between the available options

Reply:
yeah i suppose I thought of it differently because of the unset behavior.  With required it implies there is the option to not have it selected, and then manually select it again, however here we remove the option to deselect entirely....requiring the selection to be present...and now i've changed my mind lol. Changing it to required

Reply:
FWIW, I would look at whatever we do in select component for consistency. This is just a specific cased version of that.

Reply:
Looked however it appears it doesn't have a required behavior option. its normally handled by the parent with validation, it appears its just this component with the special behavior

Reply:
It has `showClearSelected` which is kind of equivalent to what you have - the one side of validation to prevent _reaching_ an empty state, with validator behavior handled in the parent. Do with that what you will!